lapce-data/src/data.rs

```diff
@@ -275,6 +320,135 @@ impl LapceData {
         }
     }
 
+    /// Parses a path into file component and LineCol (line + column combined) components
+    pub fn parse_path_line_column(path: &Path) -> Option<(PathBuf, Option<LineCol>)> {
+        // Convert path to str if path is UTF-8 valid
+        if let Some(str) = path.to_str() {
+            // Path components separated by :
+            let mut splits = str.rsplit(':');
+            // First right component
+            if let Some(first_rhs) = splits.next() {
+                // Is the first right component a number?
+                match first_rhs.parse::<usize>() {
+                    // First right component is a number
+                    Ok(first_rhs_num) => {
+                        // Second right component
+                        if let Some(second_rhs) = splits.next() {
+                            // Is the second right component a number?
+                            match second_rhs.parse::<usize>() {
+                                // Second right component is a number
+                                Ok(second_rhs_num) => { 
+                                    // Interpret [all components to the left of rhs2] combined as path
+                                    let path_str:String = splits.rev().map(|s| s.to_string()).collect::<String>();
```

Comment:
1. This is definitely not right. You can't just use an iterator here, because then you will mangle the original path (the iterator will strip out all characters that are ":". Roughly speaking, you will turn this path "C:\lapce\file.txt:45:45" into "C\lapce\file.txt". The closest correct variant would be: splits.rev().collect::<Vec<_>>().join(":") - but please do not use it, better take my second function;
2. This solution uses a lot of intermediate allocations. This is not rational in terms of performance. Why extra allocations when they can be avoided?
3. You have not converted my code quite correctly. In particular, you missed the parsing of this path "C:\lapce\file:666:45" where "C:\lapce\file:666" is the name of the file without the file extension (ends with Number of the beast, why not), 45 is the line;

Reply:
And I also noticed that you also threw out the check that the entire path is a file.  That is, I will not be able to parse this file: "C:\lapce\file_05:01:2023" - which is a perfectly valid name for the file (the extension may be missed)